src/coreclr/nativeaot/Runtime/DebugHeader.cpp

```diff
@@ -0,0 +1,294 @@
+    MAKE_DEBUG_ENTRY(EEType, m_pBaseType, offsetof(EEType, m_RelatedType) + offsetof(EEType::RelatedTypeUnion, m_pBaseType));
+    MAKE_DEBUG_ENTRY(EEType, m_ppBaseTypeViaIAT, offsetof(EEType, m_RelatedType) + offsetof(EEType::RelatedTypeUnion, m_ppBaseTypeViaIAT));
+    MAKE_DEBUG_ENTRY(EEType, m_pCanonicalType, offsetof(EEType, m_RelatedType) + offsetof(EEType::RelatedTypeUnion, m_pCanonicalType));
+    MAKE_DEBUG_ENTRY(EEType, m_ppCanonicalTypeViaIAT, offsetof(EEType, m_RelatedType) + offsetof(EEType::RelatedTypeUnion, m_ppCanonicalTypeViaIAT));
+    MAKE_DEBUG_ENTRY(EEType, m_pRelatedParameterType, offsetof(EEType, m_RelatedType) + offsetof(EEType::RelatedTypeUnion, m_pRelatedParameterType));
+    MAKE_DEBUG_ENTRY(EEType, m_ppRelatedParameterTypeViaIAT, offsetof(EEType, m_RelatedType) + offsetof(EEType::RelatedTypeUnion, m_ppRelatedParameterTypeViaIAT));
+    MAKE_DEBUG_FIELD_ENTRY(EEType, m_VTable);
+
+    MAKE_SIZE_ENTRY(StressLog);
+    MAKE_DEBUG_FIELD_ENTRY(StressLog, facilitiesToLog);
+    MAKE_DEBUG_FIELD_ENTRY(StressLog, levelToLog);
+    MAKE_DEBUG_FIELD_ENTRY(StressLog, totalChunk);
+    MAKE_DEBUG_FIELD_ENTRY(StressLog, logs);
+    MAKE_DEBUG_FIELD_ENTRY(StressLog, tickFrequency);
+    MAKE_DEBUG_FIELD_ENTRY(StressLog, startTimeStamp);
+    MAKE_DEBUG_FIELD_ENTRY(StressLog, startTime);
+    MAKE_DEBUG_FIELD_ENTRY(StressLog, moduleOffset);
+
+    MAKE_SIZE_ENTRY(ThreadStressLog);
+    MAKE_DEBUG_FIELD_ENTRY(ThreadStressLog, next);
+    MAKE_DEBUG_FIELD_ENTRY(ThreadStressLog, threadId);
+    MAKE_DEBUG_FIELD_ENTRY(ThreadStressLog, isDead);
+    MAKE_DEBUG_FIELD_ENTRY(ThreadStressLog, readHasWrapped);
+    MAKE_DEBUG_FIELD_ENTRY(ThreadStressLog, writeHasWrapped);
+    MAKE_DEBUG_FIELD_ENTRY(ThreadStressLog, curPtr);
+    MAKE_DEBUG_FIELD_ENTRY(ThreadStressLog, readPtr);
+    MAKE_DEBUG_FIELD_ENTRY(ThreadStressLog, chunkListHead);
+    MAKE_DEBUG_FIELD_ENTRY(ThreadStressLog, chunkListTail);
+    MAKE_DEBUG_FIELD_ENTRY(ThreadStressLog, curReadChunk);
+    MAKE_DEBUG_FIELD_ENTRY(ThreadStressLog, curWriteChunk);
+    MAKE_DEBUG_FIELD_ENTRY(ThreadStressLog, chunkListLength);
+    MAKE_DEBUG_FIELD_ENTRY(ThreadStressLog, pThread);
+    MAKE_DEBUG_FIELD_ENTRY(ThreadStressLog, origCurPtr);
+
+    MAKE_SIZE_ENTRY(StressLogChunk);
+    MAKE_DEFINE_ENTRY(StressLogChunk_ChunkSize, STRESSLOG_CHUNK_SIZE);
+    MAKE_DEBUG_FIELD_ENTRY(StressLogChunk, prev);
+    MAKE_DEBUG_FIELD_ENTRY(StressLogChunk, next);
+    MAKE_DEBUG_FIELD_ENTRY(StressLogChunk, buf);
+    MAKE_DEBUG_FIELD_ENTRY(StressLogChunk, dwSig1);
+    MAKE_DEBUG_FIELD_ENTRY(StressLogChunk, dwSig2);
+
+    MAKE_SIZE_ENTRY(StressMsg);
+    MAKE_DEBUG_FIELD_ENTRY(StressMsg, fmtOffsCArgs);
+    MAKE_DEBUG_FIELD_ENTRY(StressMsg, facility);
+    MAKE_DEBUG_FIELD_ENTRY(StressMsg, timeStamp);
+    MAKE_DEBUG_FIELD_ENTRY(StressMsg, args);
+
+    MAKE_SIZE_ENTRY(Object);
+    MAKE_DEBUG_FIELD_ENTRY(Object, m_pEEType);
+
+    MAKE_SIZE_ENTRY(Array);
+    MAKE_DEBUG_FIELD_ENTRY(Array, m_Length);
```

Comment:
I do not think we need to have the System.Object and array layout configurable. They are pretty much set in stone.